buildsystem/python.cmake

```diff
@@ -1,381 +1,52 @@
-
-	# filled by pxdgen and add_pxd. for use as depends list for cythonize.py.
-	set_property(GLOBAL PROPERTY SFT_PXD_FILES)
-	# filled by pxdgen. for use as output list for the pxdgen target.
-	set_property(GLOBAL PROPERTY SFT_GENERATED_PXD_FILES)
-
-	# filled with all .py filenames; used for installing.
-	set_property(GLOBAL PROPERTY SFT_PY_FILES)
-
-	# filled with the relative source dirs of the above py files.
-	set_property(GLOBAL PROPERTY SFT_PY_FILES_RELSRCDIRS)
-
-	# filled with all .py filenames that should not be installed
-	set_property(GLOBAL PROPERTY SFT_PY_FILES_NOINSTALL)
-
-	# filled with all cython module target names; used for in-place creation.
-	set_property(GLOBAL PROPERTY SFT_CYTHON_MODULE_TARGETS)
-endfunction()
-
-
-function(add_cython_modules)
-	# adds a new module for cython compilation, by relative filename.
-	# synoposis:
-	# add_cython_modules(
-	#    test.pyx
-	#    EMBED __main__.pyx
-	#    STANDALONE EMBED foo/bar.pyx
-	#    NOINSTALL STANDALONE foo/test.pyx
-	# )
-	#
-	# test.pyx is compiled to a shared library linked against PYEXT_LINK_LIBRARY
-	# __main__.pyx is compiled to a executable with embedded python interpreter,
-	# linked against libpython and PYEXT_LINK_LIBRARY.
-	# foo/bar.pyx is compiled to a executable with embedded pytthon interpreter,
-	# linked only against libpython.
-	# foo/test.pyx is compiled to a shared library linked against nothing, and will
-	# not be installed.
-
-	file(RELATIVE_PATH REL_CURRENT_SOURCE_DIR
-		"${CMAKE_SOURCE_DIR}"
-		"${CMAKE_CURRENT_SOURCE_DIR}")
-
-	set(EMBED_NEXT FALSE)
-	set(STANDALONE_NEXT FALSE)
-	set(NOINSTALL_NEXT FALSE)
-	foreach(source ${ARGN})
-		if(source STREQUAL "EMBED")
-			set(EMBED_NEXT TRUE)
-		elseif(source STREQUAL "STANDALONE")
-			set(STANDALONE_NEXT TRUE)
-		elseif(source STREQUAL "NOINSTALL")
-			set(NOINSTALL_NEXT TRUE)
-		else()
-			if(NOT IS_ABSOLUTE "${source}")
-				set(source "${CMAKE_CURRENT_SOURCE_DIR}/${source}")
-			endif()
-
-			if(NOT "${source}" MATCHES ".*\\.pyx?$")
-				message(FATAL_ERROR "non-.py/.pyx file given to add_cython_modules: ${source}")
-			endif()
-
-			get_filename_component(OUTPUTNAME "${source}" NAME_WE)
-			string(REGEX REPLACE "\\.pyx?$" ".cpp" CPPNAME "${source}")
-			set_source_files_properties("${CPPNAME}" PROPERTIES GENERATED ON)
-
-			# construct some hopefully unique target name
-			file(RELATIVE_PATH TARGETNAME "${CMAKE_SOURCE_DIR}" "${source}")
-			string(REGEX REPLACE "\\.pyx?$" "" TARGETNAME "${TARGETNAME}")
-			string(REGEX REPLACE "/" "_" TARGETNAME "${TARGETNAME}")
-
-			# generate the pretty module name
-			file(RELATIVE_PATH PRETTY_MODULE_NAME "${CMAKE_SOURCE_DIR}" "${source}")
-			string(REGEX REPLACE "\\.pyx?$" "" PRETTY_MODULE_NAME "${PRETTY_MODULE_NAME}")
-			string(REGEX REPLACE "/" "." PRETTY_MODULE_NAME "${PRETTY_MODULE_NAME}")
-			set(PRETTY_MODULE_PROPERTIES "")
-
-			if(EMBED_NEXT)
-				set(PRETTY_MODULE_PROPERTIES "${PRETTY_MODULE_PROPERTIES} [embedded interpreter]")
```

Comment:
I am not handling embedding as of now (haven't figured it out), as it is not used anywhere except for run executable.